File: robocode.samples/src/main/java/sample/MyFirstJuniorRobot.java
/*
 * Copyright (c) 2001-2025 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package sample;


import robocode.JuniorRobot;


/**
 * MyFirstJuniorRobot - a sample robot that demonstrates basic movement patterns and targeting.
 * <p>
 * Moves in a seesaw motion and spins the gun around at each end
 * when it cannot see any enemy robot. When the robot sees an enemy
 * robot, it will immediately turn the gun and fire at it.
 *
 * @author Flemming N. Larsen (original)
 */
public class MyFirstJuniorRobot extends JuniorRobot {

	/**
	 * The main run method - Implements a seesaw movement pattern
	 */
	public void run() {
		// Set robot colors (body, gun, radar)
		setColors(green, black, blue);

		// Seesaw movement pattern - repeats forever
		while (true) {
			ahead(100); // Move forward 100 pixels
			turnGunRight(360); // Scan for enemies by spinning gun 360 degrees
			back(100); // Move backward 100 pixels
			turnGunRight(360); // Scan for enemies by spinning gun 360 degrees
		}
	}

	/**
	 * Called when our robot's radar detects another robot
	 * Responds by aiming the gun at the detected robot and firing
	 */
	public void onScannedRobot() {
		// Aim the gun directly at the detected robot
		turnGunTo(scannedAngle);

		// Fire with power level 1 (low power)
		fire(1);
	}

	/**
	 * Called when our robot is hit by a bullet
	 * Responds by moving perpendicular to the bullet's path
	 * to potentially avoid future shots from the same direction
	 */
	public void onHitByBullet() {
		// Move forward 100 pixels while turning left perpendicular to the bullet's path
		turnAheadLeft(100, 90 - hitByBulletBearing);
	}
}

File: plugins/dotnet/robocode.dotnet.samples/src/SampleCs/MyFirstJuniorRobot.cs
/**
 * Copyright (c) 2001-2019 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
using Robocode;

namespace SampleCs
{
    /// <summary>
    ///   MyFirstJuniorRobot - a sample robot by Flemming N. Larsen
    ///   <p />
    ///   Moves in a seesaw motion, and spins the gun around at each end
    ///   when it cannot see any enemy robot. When the robot sees and enemy
    ///   robot, it will immediately turn the gun and Fire at it.
    /// </summary>
    public class MyFirstJuniorRobot : JuniorRobot
    {
        /// <summary>
        ///   MyFirstJuniorRobot's run method - Seesaw as default
        /// </summary>
        public override void Run()
        {
            // Set robot colors
            SetColors(GREEN, BLACK, BLUE);

            // Seesaw forever
            while (true)
            {
                Ahead(100); // Move ahead 100
                TurnGunRight(360); // Spin gun around
                Back(100); // Move back 100
                TurnGunRight(360); // Spin gun around
            }
        }

        /// <summary>
        ///   When we see a robot, turn the gun towards it and Fire
        /// </summary>
        public override void OnScannedRobot()
        {
            // Turn gun to point at the scanned robot
            TurnGunTo(ScannedAngle);

            // Fire!
            Fire(1);
        }

        /// <summary>
        ///   We were hit!  Turn and move perpendicular to the bullet,
        ///   so our seesaw might avoid a future shot.
        /// </summary>
        public override void OnHitByBullet()
        {
            // Move ahead 100 and in the same time turn left papendicular to the bullet
            TurnAheadLeft(100, 90 - HitByBulletBearing);
        }
    }
}